src/GraphQL/Execution/SubscriptionExecutionStrategy.cs

```diff
@@ -0,0 +1,296 @@
+using GraphQL.Subscription;
+using GraphQL.Types;
+using GraphQLParser.AST;
+
+namespace GraphQL.Execution
+{
+    /// <summary>
+    /// Executes a subscription.
+    /// </summary>
+    public class SubscriptionExecutionStrategy : ExecutionStrategy
+    {
+        private readonly IExecutionStrategy _baseExecutionStrategy;
+
+        /// <summary>
+        /// Initializes a new instance with a parallel execution strategy for child nodes.
+        /// </summary>
+        public SubscriptionExecutionStrategy()
+            : this(new ParallelExecutionStrategy())
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance with the specified execution strategy for child nodes.
+        /// </summary>
+        public SubscriptionExecutionStrategy(IExecutionStrategy baseExecutionStrategy)
+        {
+            _baseExecutionStrategy = baseExecutionStrategy ?? throw new ArgumentNullException(nameof(baseExecutionStrategy));
+        }
+
+        /// <summary>
+        /// Gets a static instance of <see cref="SubscriptionExecutionStrategy"/>.
+        /// </summary>
+        public static SubscriptionExecutionStrategy Instance { get; } = new();
+
+        /// <summary>
+        /// Executes a GraphQL subscription request and returns the result. The result consists
+        /// of one or more streams of GraphQL responses, returned within <see cref="SubscriptionExecutionResult.Streams"/>.
+        /// No serializable <see cref="ExecutionResult"/> is directly returned.
+        /// </summary>
+        public override async Task<ExecutionResult> ExecuteAsync(ExecutionContext context)
+        {
+            var rootType = GetOperationRootType(context);
+            var rootNode = BuildExecutionRootNode(context, rootType);
+
+            var streams = await ExecuteSubscriptionNodesAsync(context, rootNode.SubFields!).ConfigureAwait(false);
+
+            // if execution is successful, errors and extensions are not returned per the graphql-ws protocol
+            // if execution is unsuccessful, the DocumentExecuter will add context errors to the result
+
+            return new SubscriptionExecutionResult(context)
```

Comment:
So... initial _data_ in execution result for subscriptions is `null`, right?

Reply:
correct.